pkg/nodeconfig/nodeconfig.go

```diff
@@ -289,20 +290,41 @@ func (nc *nodeConfig) getWICDServiceAccountSecret() (*core.Secret, error) {
 		return nil, err
 	}
 	// Go through all the secrets in the WMCO namespace, and find the token secret which contains the auth credentials
-	// for the WICD ServiceAccount. This secret's name will always have the form:
-	// ${service_account_name}-token-${random_string}.
-	tokenSecretPrefix := "windows-instance-config-daemon-token-"
+	// for the WICD ServiceAccount
 	var filteredSecrets []core.Secret
 	for _, secret := range secrets.Items {
-		if strings.HasPrefix(secret.Name, tokenSecretPrefix) {
+		if secret.Type != core.SecretTypeServiceAccountToken {
+			// skip non-serviceAccount token secrets
+			continue
+		}
+		if secret.Annotations[core.ServiceAccountNameKey] == windows.WicdServiceName {
 			filteredSecrets = append(filteredSecrets, secret)
 		}
 	}
-	if len(filteredSecrets) != 1 {
-		return nil, fmt.Errorf("expected 1 secret with '%s' prefix, found %d", tokenSecretPrefix, len(filteredSecrets))
+	if len(filteredSecrets) == 1 {
+		return &filteredSecrets[0], nil
+	}
+	if len(filteredSecrets) > 1 {
+		return nil, fmt.Errorf("expected 1 secret for SA '%s', found %d", windows.WicdServiceName, len(filteredSecrets))
+	}
+	// no secret token found for WICD service account, create one
+	if err := nc.createWICDServiceAccountToken(); err != nil {
+		return nil, err
 	}
-	return &filteredSecrets[0], nil
+	// retry the secret token for the WICD ServiceAccount
+	return nc.getWICDServiceAccountSecret()
```

Comment:
would prefer if createWICDServiceAccountToken() returned the Secret obj so this avoids recursion and turns into:
```
	saToken, err := nc.createWICDServiceAccountToken()
	if err != nil {
		return nil, err
	}
	return saToken
```

Reply:
we should also update the `getWICDServiceAccountSecret()` func doc to say it creates the token if it does not exist already

Reply:
> we should also update the `getWICDServiceAccountSecret()` func doc to say it creates the token if it does not exist already

Good point, will include it.